### DISCUSSION TITLE
Any node similar to the pointiness in cycles?

### DISCUSSION THREAD

**Discussion by HSDSZ** [bnpr/Malt#224] [Q&A]
I am planning ob some more complex procedural nodes in MALT.

Any node or work around or code that can achieve the pointiness in cycles?

**Comment by pragma37** [comment]
You can use the *curvature* and *surface_curvature* functions.
![imagen](https://user-images.githubusercontent.com/36610999/151606592-88db2581-8373-4904-a945-90794a547846.png)

**Reply by HSDSZ** [reply]
![Pointiness](https://user-images.githubusercontent.com/36077492/151654772-a95f7f1d-545d-454a-99e5-b3340bc56450.JPG)

Great, Surface curvature works. 
But the viewport and render threshold seems to be difference. I set all the preview parameters in "world" same as render. but the results are still difference.

If I want them to be the same, how should I change it?

**Reply by HSDSZ** [reply]
Besides, what does the "depth_range" input of get_surface_curvature do? I change it from negative to postive, its value has no observable influence.

**Reply by pragma37** [reply]
Surface curvature only takes into account the pixels that lies within the *depth range*.
Here you can see the difference:
![imagen](https://user-images.githubusercontent.com/36610999/151673343-38cdc8a5-324f-4373-9ddb-96bae9ec742b.png)
left = 0.1/ right = 3

The difference you see is most likely due to the resolution difference. The curvature filter checks neighbor pixels, so closer objects/high resolutions can yield different values than farther objects/lower resolutions.
I'll try to find a way to make it more stable.

Edit: The nodes look different because that's the (not yet released) 1.0 preview,  but the curvature works the same.
